### DISCUSSION TITLE
Help me with the keyboard backlight keycode

### DISCUSSION THREAD

**Discussion by RiccardoPP** [linux-surface/linux-surface#1193] [Ideas]
Hello, 
I'm running Arch Linux on SL3 AMD 15'' without issue (keyboard backlit included). Thanks for your code and support! 

Now I'm trying to add an **OSD** notification when the **keyboard backlight key** is pressed, but I can't locate his keycode. With `wev` command (or similar software) I don't have any output and I've read on other discussions that it's because that key is controlled by the Surface EC (or other internal chip).

The keycode:

`keycode 236 = XF86KbdLightOnOff NoSymbol XF86KbdLightOnOff`

obliviously don't work. Any idea how to investigate this?

**Comment by qzed** [comment]
What you read in the other posts is, as far as I understand, correct: The keyboard backlight key is wired up directly (and exclusively) to the embedded controller. This means that Linux doesn't receive any event or notification when you press the button. Similarly, you can't control the keyboard backlight via Linux.

It might be possible that I'm missing something, but as far as I can tell, what you want to do is unfortunately just not possible on those machines.

**Reply by RiccardoPP** [reply]
Ok, understood. Thank you.
